### ISSUE TITLE
Setting Firmware to EFI - and Esxi 6.7 changes

### ISSUE THREAD

**Issue by jcoconnor** [jetbrains-infra/packer-builder-vsphere#138]
All our vm's need EFI rather than the default BIOS.

I was able to fudge this by putting ```"firmware" : "efi"``` in the ```configuration_parameters```

However, the cluster has recently been upgraded to Esxi 6.7 and magically, the ```efi``` and a few other settings such as ```"windows.vbs.enabled" : "TRUE"``` are being filtered out of the ```vmx``` file when it lands on the ```vcenter```

This has broken our build pipeline which was working nicely up to this.

Suspect this is an Esxi 6.7 issue and may well be tied to [this new functionality](https://blogs.vmware.com/vsphere/2018/05/introducing-support-virtualization-based-security-credential-guard-vsphere-6-7.html), but it is a major nuisance.

Its also reminiscent of an ```ovftool``` (mis)feature which also deliberately filters out the HW-Nested virtualisation settings from the ```vmx``` file, but so far haven't found any documentation or release note that relates to this.

Is anyone aware of workarounds to this ?

**Comment by nate-duke** [comment]
We're also in a situation of requiring EFI booted VMs and using this plugin for our template builds. We haven't updated to ESXI 6.7 yet but will do in the next couple of weeks so I'm interested any fixes/workarounds regarding this issue.

I'll share anything we find if we do end up upgrading before this is addressed.

**Comment by BilboTheGreedy** [comment]
We also would like this, currently we create one manually and then use the clone method instead since with iso method you cannot select this option. Seems some code need to be added to step_hardware.go for this to be supported.

**Comment by BilboTheGreedy** [comment]
In pyvomi you at least have [this](https://github.com/vmware/pyvmomi/blob/master/docs/vim/vm/GuestOsDescriptor/FirmwareType.rst) option . So after adding like "FirmwareType" as string to HardwareConfig Struct and in dependent code etc. Option to select "efi" or "bios" would be plausible? Seems to be implemented in [govc ](https://github.com/vmware/govmomi/blob/919b728c0b389387a230e936146e23358a846ae9/govc/vm/create.go)

**Comment by jcoconnor** [comment]
I've had some discussion with our infra team who are in the process of moving all of our infra to Esx 6.7. It seems from the release notes that you need to specify the firmware when the machine is created, and not in the vmx_data, or at least that's my reading of both the behavior and this statement in the release notes:
```
Changing the boot type from legacy BIOS to UEFI after you install ESXi 6.7 might cause the host
to fail to boot. In this case, the host displays an error message similar to Not a VMware boot bank.
Changing the host boot type between legacy BIOS and UEFI is not supported after you install ESXi 6.7.
```

I was trying to go thru the code but haven't written go yet, so not familiar enough to suggest changes.

**Comment by jcoconnor** [comment]
Thanks for the pointers @BilboTheGreedy 
I have some initial changes that appear to create the VM with the selected firmware and boot it up.
Will tidy it up over the next few days and hopefully can present a PR for it.

**Comment by jcoconnor** [comment]
See #156 for initial PR - still to add docs, validation of the value and unit tests, but this appears to do the job correctly for the iso option at least.

**Comment by mkuzmin** [comment]
fixed by #156
